legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BulkLoadPlanner.java

```diff
@@ -225,25 +204,34 @@ List<String> getDigestOrRemainingColumns()
     @Override
     public LogicalPlan buildLogicalPlanForMetadataIngest(Resources resources)
     {
-        BulkLoadMetadataUtils bulkLoadMetadataUtils = new BulkLoadMetadataUtils(bulkLoadMetadataDataset);
-        String batchSourceInfo = jsonifyBatchSourceInfo(stagedFilesDataset.stagedFilesDatasetProperties());
-        StringValue datasetName = StringValue.of(mainDataset().datasetReference().name());
-        Insert insertMetaData = bulkLoadMetadataUtils.insertMetaData(datasetName, StringValue.of(batchSourceInfo));
-        return LogicalPlan.of(Arrays.asList(insertMetaData));
+        // Save file paths/patterns and event id into batch_source_info column
+        Map<String, Object> batchSourceInfoMap = new HashMap<>();
+        LogicalPlanUtils.jsonifyBulkLoadSourceInfo(batchSourceInfoMap, stagedFilesDataset.stagedFilesDatasetProperties(), options().bulkLoadEventIdValue());
```

Comment:
We can let the jsonifyBulkLoadSourceInfo return the json as a map

Reply:
Sure, changed